### DISCUSSION TITLE
Extend (stepped) Line in Composed Chart

### DISCUSSION THREAD

**Discussion by master117** [recharts/recharts#4243] [Q&A]
Hi, I have a composed chart of Stacked Bars and a stepped Line.

![grafik](https://github.com/recharts/recharts/assets/444009/c574ebef-73e3-4c92-96f0-41755c4d9931)

I would like to extend the line to the start of the graph and the end of the graph. Is this possible? Should I use something else to draw the line than a line chart?

**Comment by ckifer** [comment]
Since this is a composed, categorical chart and Bars are present the line ends at the category itself. Anything before that has no additional category to connect to, but the domain has to be extended in order to fit the bar. 

As for solutions.. I'm pretty sure this is your best bet unless you'd like to make a custom line. A stepped reference line isn't realistically possible. On mobile right now but I can try to help with a custom line later

**Reply by master117** [reply]
Thank you so far!

>As for solutions.. I'm pretty sure this is your best bet unless you'd like to make a custom line. A stepped reference line isn't realistically possible. On mobile right now but I can try to help with a custom line later

I'm not sure what you are referring to with "this is your best bet", you mean my attempt so far? Or to extend the domain somehow?
Also, of course I would be very happy about any help. :)

**Reply by ckifer** [reply]
Ah sorry, I just mean that there isn't a way to do it by default so what you have is the best you're going to get without a custom line component.

Eventually, we should maybe add a prop that extends lines to the min domain/max domain no matter what.

**Reply by ckifer** [reply]
Would you be able to throw what you currently have in a codesandbox so I can help?

**Reply by ckifer** [reply]
Here is an old issue I found with a potential but ugly workaround https://github.com/recharts/recharts/issues/1669

You could also attempt to do it with a LineChart component and then use a customized bar similar to this example https://github.com/recharts/recharts/blob/3.x/storybook/stories/API/component/Customized.stories.tsx#L37-L90

**Reply by master117** [reply]
Here is a codesandbox of the issue https://codesandbox.io/p/sandbox/jolly-heyrovsky-pd728v

**Reply by ckifer** [reply]
[![Edit elegant-nash-tlpfxr](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/elegant-nash-tlpfxr?file=%2Fsrc%2FApp.tsx%3A141%2C19-141%2C33)

Really quite ugly but it works
